src/main/java/seedu/tp/flashcard/FlashcardList.java

```diff
@@ -146,22 +151,45 @@ public FlashcardList reviewRandomFlashcards(Ui ui) throws InvalidFlashcardIndexE
             if (flashcard.isReviewed()) {
                 System.out.println("You have already reviewed this flashcard.");
                 System.out.println("");
-            } else if (ui.promptUserResponseForReviewing(flashcard).equals("yes")) {
-                ReviewedCommand reviewedCommand = new ReviewedCommand(this,
-                    flashcards.indexOf(flashcard));
-                CommandFeedback reviewedCommandFeedback = reviewedCommand.execute();
-                ui.showCommandFeedback(reviewedCommandFeedback);
-                reviewedNumber++;
             } else {
-                continue;
+                reviewedNumber = handleResponse(ui, flashcard, reviewedNumber);
             }
         }
-        totalReviewedNumber += reviewedNumber;
         int totalUnreviewedNumber = flashcards.size() - totalReviewedNumber;
         ui.confirmRandomFlashcardsReviewCompletion(reviewedNumber, totalUnreviewedNumber);
         return randomFlashcards;
     }
 
+    private int handleResponse(Ui ui, Flashcard flashcard, int reviewedNumber)
+            throws InvalidFlashcardIndexException, InvalidInputFormatException {
+        int statusCode = NORMAL_CODE;
+        do {
+            try {
+                String response = ui.promptUserResponseForReviewing(flashcard).toLowerCase();
+                switch (response) {
+                case "y":
+                case "yes":
+                    ReviewedCommand reviewedCommand = new ReviewedCommand(this,
+                            flashcards.indexOf(flashcard));
+                    CommandFeedback reviewedCommandFeedback = reviewedCommand.execute();
+                    ui.showCommandFeedback(reviewedCommandFeedback);
+                    reviewedNumber++;
+                    break;
+                case "n":
+                case "no":
+                    break;
+                default:
+                    throw new InvalidInputFormatException();
+                }
+                return reviewedNumber;
+            } catch (InvalidInputFormatException e) {
+                System.out.println("The answer given is not acceptable. Please enter again.");
```

Comment:
I think a clearer way to phrase it can simply be "Please enter either yes/no."

Reply:
> I think a clearer way to phrase it can simply be "Please enter either yes/no."

Agree!